org.eclipse.dawnsci.analysis.dataset.test/src/org/eclipse/dawnsci/analysis/dataset/FunctionalUtilsTest.java

```diff
@@ -0,0 +1,210 @@
+package org.eclipse.dawnsci.analysis.dataset;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.util.stream.IntStream;
+
+import org.eclipse.january.dataset.ByteDataset;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DoubleDataset;
+import org.eclipse.january.dataset.FloatDataset;
+import org.eclipse.january.dataset.IndexIterator;
+import org.eclipse.january.dataset.IntegerDataset;
+import org.eclipse.january.dataset.LongDataset;
+import org.eclipse.january.dataset.Random;
+import org.eclipse.january.dataset.ShortDataset;
+import org.eclipse.january.dataset.Slice;
+import org.junit.Test;
+
+public class FunctionalUtilsTest {
+
+	@Test
+	public void testDatasetIndexFunction() {
+		Dataset a = DatasetFactory.createRange(144).reshape(12, -1);
+		generateAndCheckFunction(a);
+		generateAndCheckFunction(a.getTransposedView());
+
+		Dataset v = a.getSliceView(new Slice(3, 10), new Slice(null, null, -2));
+		generateAndCheckFunction(v);
+		generateAndCheckFunction(v.getTransposedView());
+
+		Dataset z = a.getSliceView(new Slice(3, 3), new Slice(1, 1));
+		assertEquals(0, z.getSize());
+		generateAndCheckFunction(z);
+		generateAndCheckFunction(z.getTransposedView());
+	}
+
+	private int[] generateIndexes(Dataset v) {
+		int[] eIndex = new int[v.getSize()];
+		IndexIterator it = v.getIterator();
+		for (int i = 0; it.hasNext(); i++) {
+			eIndex[i] = it.index;
+		}
+		return eIndex;
+	}
+
+	private void generateAndCheckFunction(Dataset v) {
+		int[] eIndex = generateIndexes(v);
+		int[] aIndex = IntStream.range(0, v.getSize()).map(FunctionalUtils.createDatasetIndexFunction(v)).toArray();
+
+		assertArrayEquals(eIndex, aIndex);
+	}
+
+	@Test
+	public void testDatasetIndexStream() {
+		testDatasetIndexStream(false);
+		testDatasetIndexStream(true);
+	}
+
+	private void testDatasetIndexStream(boolean parallel) {
+		Dataset a = DatasetFactory.createRange(144).reshape(12, -1);
+		generateAndCheckStream(a, parallel);
+		generateAndCheckStream(a.getTransposedView(), parallel);
+
+		Dataset v = a.getSliceView(new Slice(3, 10), new Slice(null, null, -2));
+		generateAndCheckStream(v, parallel);
+		generateAndCheckStream(v.getTransposedView(), parallel);
+
+		Dataset z = a.getSliceView(new Slice(3, 3), new Slice(1, 1));
+		generateAndCheckStream(z, parallel);
+		generateAndCheckStream(z.getTransposedView(), parallel);
+	}
+
+	private void generateAndCheckStream(Dataset v, boolean parallel) {
+		int[] eIndex = generateIndexes(v);
+		int[] aIndex = FunctionalUtils.createDatasetIndexStream(v, parallel).toArray();
+
+		assertArrayEquals(eIndex, aIndex);
+	}
+
+	@SuppressWarnings("unchecked")
+	private static final Class <? extends Dataset>[] allDS = new Class[] {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)